Cabal/Distribution/Simple/PreProcess.hs

```diff
@@ -123,6 +123,10 @@ data PreProcessor = PreProcessor {
 -- preprocessor's output name format.
 type PreProcessorExtras = FilePath -> IO [FilePath]
 
+-- | A newtype around 'PreProcessorExtras', useful for storing
+-- 'PreProcessorExtras' inside of another type constructor (e.g., a list)
+-- without impredicativity.
+newtype WrappedPreProcessorExtras = WrapPPE { unWrapPPE :: PreProcessorExtras }
```

Comment:
Could you mention in the comment, that `IO` is polymorphic type. It's easy to forget (or don't know)